lib/model/shared_postcard.dart

```diff
@@ -42,12 +47,38 @@ class SharedPostcard {
     };
   }
 
+  bool get isExpired {
+    if (sharedAt == null) {
+      return false;
+    }
+    return sharedAt!
+        .add(POSTCARD_SHARE_LINK_VALID_DURATION)
+        .isBefore(DateTime.now());
+  }
+
   @override
   int get hashCode {
     return tokenID.hashCode ^ owner.hashCode;
   }
 }
 
+extension ListSharedPostcard on List<SharedPostcard> {
+  Future<List<SharedPostcard>> get expiredPostcards async {
+    final expiredPostcards = <SharedPostcard>[];
+    await Future.forEach(this, (SharedPostcard postcard) async {
```

Comment:
using `await Future.wait(
    map((SharedPostcard postcard) async {` may help to speed up a little bit. 
    
It will execute the asynchronous concurrently.